### DISCUSSION TITLE
Zombie Channel Recovery Assistance

### DISCUSSION THREAD

**Discussion by tmaiorca** [lightningnetwork/lnd#7610] [Troubleshooting] [answered]
My peer and I are having some difficulty with performing a zombie channel recovery using chantools. We have attempted this without success but I believe it may be due to our ignorance and/or lack of correct values while generating the key pair from step 4 of the documentation. 

For reference, I am using the [Zombie Channel Recovery documentation](https://github.com/guggero/chantools/blob/master/doc/zombierecovery.md), part of the [chantools repo](https://github.com/guggero/chantools).

 **Background**:
My peer and I connected over reddit and agreed to open a small channel for testing purposes. Unfortunately after a few weeks, both our nodes crashed without backup of the channel states/channel DB. We were successful with restoring on-chain funds, however restoring our SCB led us to `waiting_close_channels` (output from `lncli pendingchannels`). I previously created lnd [issue #7595](https://github.com/lightningnetwork/lnd/issues/7595) which led me to the chantools documentation and thus zombie recovery.

**Error**:
Command: `chantools zombierecovery makeoffer --node1_keys preparedkeys-1.json --node2_keys preparedkeys-2.json --feerate 15`
Error: `invalid files, channel info incorrect`

![image](https://user-images.githubusercontent.com/46833258/232939038-cbef6987-f743-4db1-b67d-1286a831d088.png)

My peer and I could use some assistance with performing the zombie channel recovery. We're starting from Step 3 of the [Zombie Channel Recovery documentation](https://github.com/guggero/chantools/blob/master/doc/zombierecovery.md):

**match.json file**
- Since the match.json file is the used to create the preparekeys.json file(s), and since those files are required to create the offer, do both my peer and I create our own match.json file to be ran with the command in step 4, or just myself if I'm assuming to be peer 1? Meaning, do we each create a separate match.json file using unique chan_point and address values?

- For the channel point value: If I'm peer 1, do I use the output index :1 (Initiator) or :0 (remote)? I assume initiator? This is in reference to running lncli pendingchannels

- For the on-chain address value: how is this different from the output payout_addr used in step 4?

<br>

**When at step 4 to create the preparedkeys.json file:**

- Does peer 2 do this using their own unique-value match.json file, or using the one I provide to them? 

- I'm assuming since we each run the "chantools zombierecovery preparekeys --payout_addr bc1xxx --match_file /tmp/match.json" command, we replace the --payout_addr with our own on-chain address? 

<br>
<br>
This is what the channel looks like from my side after about a month...

![image](https://user-images.githubusercontent.com/46833258/232940529-339e2a19-8f1f-4b68-b8c4-b2d934078cd8.png)

**Comment by guggero** [comment, marked as answer]
I enabled the ["discussions" ](https://github.com/guggero/chantools/discussions) section in the `chantools` repo. So please use that in the future (this is already your second entry created in the wrong repo...

> do both my peer and I create our own match.json file to be ran with the command in step 4

No, you both use the the same file. Since you have two channels, you add both of them as an entry each in the `channels` section.

> For the channel point value: If I'm peer 1, do I use the output index :1 (Initiator) or :0 (remote)? I assume initiator? This is in reference to running lncli pendingchannels

The channel point is not peer dependent. It's the location of the funding transaction, so it's unique to the channel. It's exactly the value from your `lncli pendingchannels` output. But as mentioned above, if there are two channels, you need to create two entries.

> For the on-chain address value: how is this different from the output payout_addr used in step 4?

The on-chain address is the funding address of the channel. So basically it's the address that corresponds to the output of the channel point. So it would be `bc1qaug9u98vqkg32wkrwk3xkjhdg3q6sk57dzhafmjkk4rtnl3g8fssnajx7u` for the `c2c5e72c007f87325e633233d4c0d27a29dd90a9dac6755333f4d0504383b42a:0` channel. You find that on a block explorer like blockstream.info and the `:0` or `:1` in the channel point refers to the number of the output, starting at zero.

The `payout_addr` is where your part of the funds will be sent to. So you take an address from your wallet and your peer chooses a different address.

I think it's probably easiest if you post the `match.json` here and I can tell you if it looks right. But please no screenshots, you cannot copy values from them so I can't look up things easily. You can use the triple back ticks on GitHub to format something as code.

**Reply by tmaiorca** [reply]
Thank you Guggero! I apologize for having started this discussion in the wrong repo. Going forward, I will do my due diligence to check that I'm opening an issue/discussion under the appropriate repo.

I am going to recreate the match.json file this weekend and provide that here so that you can validate its content. Once I know that is correct, I will perform the zombie recovery again with my peer and revert back with any issues we may have.

**Reply by tmaiorca** [reply]
@guggero 
> 
> I think it's probably easiest if you post the `match.json` here and I can tell you if it looks right. But please no screenshots, you cannot copy values from them so I can't look up things easily. You can use the triple back ticks on GitHub to format something as code.

I believe I have the formatting of the match.json file correct now. Would you be able to give it a look and confirm if it looks okay? The rest of the steps now make sense. I appreciate your help in getting back to me. 

```
{
    "node1": {
        "identity_pubkey": "0357a02133bf4a49e222b6cb66f894c3b4878690b0d5310a3916ffa831669d19a2",
        "contact": "peer 1 contact info"
    },
    "node2": {
        "identity_pubkey": "039910a225afe022698e41afdc69ef7b23832cb03baf632476b42d1b7ccc596fcf",
        "contact": "peer 2 contact info"
    },
    "channels": [
        {
            "short_channel_id": "850067624019296256",
            "chan_point": "c2c5e72c007f87325e633233d4c0d27a29dd90a9dac6755333f4d0504383b42a:0",
            "address": "bc1qaug9u98vqkg32wkrwk3xkjhdg3q6sk57dzhafmjkk4rtnl3g8fssnajx7u",
            "capacity": 110000
        },
	{
            "short_channel_id": "850063225917997057",
            "chan_point": "48c9fc5606311257c24e2310ee766b28ac1054da6d824c807707422034cd914e:1",
            "address": "bc1q2fflxz4890v4ect8xd45ht40u2x04uswvpfztyxgewprxlayhmyqqyg7k8",
            "capacity": 110000
        }
    ]
}

**Reply by guggero** [reply]
Yes, that looks good!

**Reply by tmaiorca** [reply]
Thanks for confirming! I've provided the match.json and preparedkey files from my end to my peer. Once I receive the PSBT from my peer I will sign and broadcast that to the network, and subsequently close this discussion.
